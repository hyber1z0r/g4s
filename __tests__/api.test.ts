jest.mock('node-fetch');

import fetch from 'node-fetch'
import API from '../src/api'
const {Â Response } = jest.requireActual('node-fetch');

test('should invoke disarm command', async () => {
	(fetch as unknown as jest.Mock).mockResolvedValue(new Response(undefined, { status: 200}))

	const api = new API('username', 'password')
	await api.disarmPanel('panelId-123')
	expect(fetch).toHaveBeenCalled();
	expect(fetch).toMatchSnapshot();
})


test('should invoke arm command', async () => {
	(fetch as unknown as jest.Mock).mockResolvedValue(new Response(undefined, { status: 200}))

	const api = new API('username', 'password')
	await api.armPanel('panelId-123')
	expect(fetch).toHaveBeenCalled();
	expect(fetch).toMatchSnapshot();
})

