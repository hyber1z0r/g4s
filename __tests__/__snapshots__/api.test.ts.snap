// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should invoke arm command 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "https://mit.g4severhome.dk/ESI.API/API/Commands/invokeAPI",
      Object {
        "body": "{\\"email\\":\\"username\\",\\"password\\":\\"password\\",\\"panelId\\":\\"panelId-123\\",\\"methodToInvoke\\":\\"Arm\\"}",
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`should invoke disarm command 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "https://mit.g4severhome.dk/ESI.API/API/Commands/invokeAPI",
      Object {
        "body": "{\\"email\\":\\"username\\",\\"password\\":\\"password\\",\\"panelId\\":\\"panelId-123\\",\\"methodToInvoke\\":\\"Disarm\\"}",
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
